### PR TITLE
Added compatibility for XNLI

### PR DESCRIPTION
**Pull request by lintangsutawika**
added template_config_name arg so that the dataset and prompt template source can be different, example: prompt from XNLI En but data to eval from XNLI Fr

### CODE REVIEW

**Comment by lintangsutawika** [comment]
Example to run eval on XNLI
```
CHECKPOINT_PATH="bigscience/bloom-350m"
OUTPUT_DIR="bloom-xnli"
dataset_name="xnli"
template_config_name="en"
dataset_config_name="fr"

python t-zero/evaluation/run_eval.py \
        --dataset_name $dataset_name \
        --dataset_config_name $dataset_config_name \
        --template_config_name $template_config_name \
        --model_name_or_path $CHECKPOINT_PATH \
        --output_dir $OUTPUT_DIR \
        --template_name 'GPT-3 style'
```

@thomasw21

**Review by VictorSanh — APPROVED** [review]
thank you @lintangsutawika! lgtm! feel free to merge!

File: evaluation/run_eval.py [hunk 1]
```diff
@@ -60,6 +60,12 @@ def parse_args():
         default=None,
         help="The configuration name of the dataset to use (via the datasets library).",
     )
+    parser.add_argument(
+        "--template_config_name",
+        type=str,
+        default=None,
+        help="The name of the dataset_config_name of the template we want to use, example: use XNLI En prompts for XNLI Fr",
```
Comment (thomasw21) [inline evaluation/run_eval.py:67]: Is using english prompts on the french XNLI something good? Like I understand it's some sort of measure multilinguality, but I would have expected to have a bunch of frnech prompts for the XNLI fr ...

**Comment by thomasw21** [comment]
Looks good to me besides the fact that prompts are english prompts (I expected to have them in their languages).

**Comment by VictorSanh** [comment]
> Looks good to me besides the fact that prompts are english prompts (I expected to have them in their languages).

i think this is fine. A bunch of the prompts coming from the eval hackathon are code switching

File: evaluation/run_eval.py (hunk 1 above)
Comment (lintangsutawika) [reply evaluation/run_eval.py:67]: I thought using english prompts for XNLI is indeed what we wanted to accomplish?

Anyway, evaluating XNLI with prompts and dataset in the same language only requires the Promptsource part be updated. So `--template_config_name` adds more flexibility while not needing to change too much of the code to be able to eval on multiple multilingual settings.

**Comment by thomasw21** [comment]
Thanks @lintangsutawika

**Pull request merged**